### DISCUSSION TITLE
Why use Kratos over custom solution?

### DISCUSSION THREAD

**Discussion by otyang** [ory/kratos#2675] [Q&A]
Hello All,

I have been trying to wrap my head on the title. Reason been, the kratos documentation is hard enough for me to understand. I have implemented my own auth in past and don't want to go that route now. 

I really love kratos from reviews and all on the internet but there are absolutely zero to nil good tutorials on implementation. So i ask if its wise for me spending time on someone else code than just using something easy else.

Thanks

**Comment by vinckr** [comment]
Hello @otyang ,
can you share a bit about what your requirements are and what alternatives you had in mind? 
Implementing basic login is probably not too complex, but verification flows, MFA, user management, identity models, social logins, passwordless can blow up the scope of your authentication project pretty quickly. After all I assume you don't want to work on authentication services but on something else that just uses a proven authentication service. It also depends on what threat/security model and scale you have in mind for your project. If you want to make sure there is no sensitive information leaked, you have to keep a lot of attack vectors in mind when designing your flows. 
Of course nothing is ever perfect, but a huge community of security specialists and other developers run Ory software in production, test the security extensively, and work on new features daily.
You can run Kratos in the managed Ory Cloud service or host it on your own infrastructure.
